### DISCUSSION TITLE
Upgrade dropwizard-jdbi3 dependency to use latest jdbi3-core dependency

### DISCUSSION THREAD

**Discussion by lohitpradhan** [dropwizard/dropwizard#5608] [General]
We use dropwizard-jdbi3 dependency of version 2.0.29 for database operations and internally it uses jdbi3-core dependency of version 3.28.0. We were facing some issue so we filed a ticket to jdbi community for fixing the issue. After investigation they found out a bug and fixed the issue in the jdbi3-core library and merged their changes to master branch. So we would like to uptake the latest changes they made in the jdbi3-core library. could you please upgrade dropwizard-jdbi3 dependency so that it uses the latest jdbi3-core (3.32.0) ?
jdbi bug link : https://github.com/jdbi/jdbi/issues/2065

**Comment by joschi** [comment]
@lohitpradhan You can bump the dependency yourself in your application, either as a direct dependency in the `<dependencies>` section of your POM or by importing [`org.jdbi:jdbi3-bom:3.32.0`](https://search.maven.org/artifact/org.jdbi/jdbi3-bom/3.32.0/pom) in the `<dependencyManagement>` section of your POM.

**Reply by lohitpradhan** [reply]
Sure. Let me try that. Thanks

**Comment by lohitpradhan** [comment]
Hi @joschi, Most of the project teams in our org are using dropwizard-jdbi3 dependency and it would be beneficial if you uptake the latest jdbi3-core changes. We are advised not to use direct jdbi3-core dependency in our project. So, kindly uptake the latest changes also may i know when is your next planned release date ? 
Thanks

**Reply by sleberknight** [reply]
On projects I'm working on, we use @joschi 's suggestion and import the JDBI BOM. We're currently using JDBI 3.32.0 and Dropwizard 2.1.1 and they work fine together. It's also good to import the BOM, if available, so that all the various versions of the library are the same, e.g. every `jdbi-xxx` is locked to the same version. For this same reason, we also import the Jackson and Jetty BOMs, so that all the various `jackson-xxx` and `jetty-xxx` are on the same version. I assume this is why someone doesn't want you to specific `jdbi3-core` directly, since then you would be using an updated version of that but different versions of the other `jdbi-xxx` JARs. But by importing the BOM, that problem is solved.

More generically, you can always use the dependency management mechanism to lock any library to a specific version, and I've found it is _usually_ safe to update minor and patch versions. For example, we were using Jersey 2.35 on Dropwizard 2.0.x well before Dropwizard 2.1 moved to that Jersey version (we took care to handle the problems with [Optional<T>](https://www.dropwizard.io/en/latest/manual/upgrade-notes/upgrade-notes-2_1_x.html#upgrade-to-jersey-2-35) parameters ourselves). But this lets you always stay on the latest patch versions of dependencies.
